src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3012,11 +3022,12 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
             type = ASRUtils::TYPE(ASR::make_Allocatable_t(al, type->base.loc,
                 ASRUtils::type_get_past_pointer(type)));
         }
-        bool is_c_p_pointer_call_copy = is_c_p_pointer_call;
-        ASR::expr_t *value = nullptr;
+
+        create_add_variable_to_scope(var_name, type,
```

Comment:
The `c_p_pointer()` needs the variable to which it is being assigned. Therefore, in this PR, we first add the variable to scope, then visit the value/`c_p_pointer()` and later set the variables `init_val` in `process_variable_init_val()`.